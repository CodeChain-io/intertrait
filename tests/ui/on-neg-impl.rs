use intertrait::*;

struct Data;

#[cast_to]
impl !Sync for Data {}

fn main() {
    let data = Data;
    let _: &dyn Sync = &data;
}
