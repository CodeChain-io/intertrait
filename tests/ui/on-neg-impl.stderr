error: #[cast_to] is not for !Trait impl
 --> $DIR/on-neg-impl.rs:6:6
  |
6 | impl !Sync for Data {}
  |      ^

error[E0658]: negative trait bounds are not yet fully implemented; use marker types for now
 --> $DIR/on-neg-impl.rs:6:1
  |
6 | impl !Sync for Data {}
  | ^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: for more information, see https://github.com/rust-lang/rust/issues/13231

error[E0277]: `Data` cannot be shared between threads safely
  --> $DIR/on-neg-impl.rs:10:24
   |
10 |     let _: &dyn Sync = &data;
   |                        ^^^^^ `Data` cannot be shared between threads safely
   |
   = help: the trait `std::marker::Sync` is not implemented for `Data`
   = note: required for the cast to the object type `dyn std::marker::Sync`
